server/apps/packages/package_service.py

```diff
@@ -266,8 +317,8 @@ def remove_refs_in_package(self, package, ref_id_to_remove, processed_packages=N
             else:
                 updates[SEQUENCE] = new_sequence
                 last_take_group = next(reference for reference in
-                                       next(new_group.get('refs') for new_group in new_groups if
-                                            new_group['id'] == 'main')
+                                       next(new_group.get(ASSOCIATIONS) for new_group in new_groups if
+                                            new_group[GROUP_ID] == MAIN_GROUP)
```

Comment:
I think a package has always groups. This is pretty much the main type:composite requirement afaik.